plugin/restore/mysql/mysql.go

```diff
@@ -50,9 +52,62 @@ func New(driver *mysql.Driver, instance *mysqlutil.Instance, connCfg db.Connecti
 	}
 }
 
-// replayBinlog restores the database using incremental backup in time range of [config.Start, config.End).
-func replayBinlog(ctx context.Context, config api.BinlogInfo, binlogDir string) error {
-	return fmt.Errorf("Unimplemented")
+// ReplayBinlog replays the binlog about `originDatabase` from `startBinlogInfo.Position` to `targetTs`.
+func (r *Restore) replayBinlog(ctx context.Context, originalDatabase, pitrDatabase string, startBinlogInfo api.BinlogInfo, targetTs int64) error {
+	if err := r.fetchBinlogUpToTargetTs(ctx); err != nil {
+		return err
+	}
+
+	replayBinlogPaths, err := getReplayBinlogPathList(startBinlogInfo, r.binlogDir)
+	if err != nil {
+		return err
+	}
+
+	// Converts the ts to the local time, which is needed by mysqlbinlog's --stop-datetime option.
+	stopTime := time.Unix(targetTs, 0)
+
+	mysqlbinlogArgs := []string{
+		"--disable-log-bin",
+		fmt.Sprintf(`--rewrite-db=%s->%s`, originalDatabase, pitrDatabase),
+		fmt.Sprintf("--database=%s", pitrDatabase),
+		fmt.Sprintf("--start-position=%d", startBinlogInfo.Position),
+		fmt.Sprintf(`--stop-datetime=%d-%d-%d %d:%d:%d`, stopTime.Year(), stopTime.Month(), stopTime.Day(), stopTime.Hour(), stopTime.Minute(), stopTime.Second()),
+	}
+
+	mysqlbinlogArgs = append(mysqlbinlogArgs, replayBinlogPaths...)
+
+	mysqlArgs := []string{
+		fmt.Sprintf("--host=%s", r.connCfg.Host),
+		fmt.Sprintf("--port=%s", r.connCfg.Port),
+		fmt.Sprintf("--user=%s", r.connCfg.Username),
+		fmt.Sprintf("--password=%s", r.connCfg.Password),
+	}
+
+	mysqlbinlogCmd := exec.CommandContext(ctx, r.mysqlutil.GetPath(mysqlutil.MySQLBinlog), mysqlbinlogArgs...)
+	mysqlCmd := exec.CommandContext(ctx, r.mysqlutil.GetPath(mysqlutil.MySQL), mysqlArgs...)
+
+	mysqlRead, err := mysqlbinlogCmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("cannot get mysqlbinlog stdout pipe, error: %w", err)
+	}
+	defer mysqlRead.Close()
+
+	mysqlbinlogCmd.Stderr = os.Stderr
+
+	mysqlCmd.Stderr = os.Stderr
+	mysqlCmd.Stdout = os.Stderr
+	mysqlCmd.Stdin = mysqlRead
+
+	if err := mysqlbinlogCmd.Start(); err != nil {
+		return fmt.Errorf("cannot start mysqlbinlog command %s, error: %w", mysqlbinlogCmd.String(), err)
+	}
+	if err := mysqlCmd.Run(); err != nil {
+		return fmt.Errorf("cannot run mysql command %s, error: %w", mysqlCmd.String(), err)
```

Comment:
```suggestion
		return fmt.Errorf("mysql command %s fails, error: %w", mysqlCmd.String(), err)
```
Maybe we run it successfully, but it crashes in the middle.